python/hail/expr/functions.py

```diff
@@ -2661,6 +2661,30 @@ def set(collection):
     return collection._method("toSet", tset(collection.dtype.element_type))
 
 
+@typecheck(t=Type)
+def empty_set(t):
+    """Returns an empty set of elements of a type `t`.
+
+    Examples
+    --------
+    .. doctest::
+
+        >>> hl.eval_expr(empty_set())
+        set()
+
+    Parameters
+    ----------
+    t : :class:`.Type`
+        Type of the set elements.
+
+    Returns
+    -------
+    :class:`.SetExpression`
+        Empty set of elements of type `t`.
+    """
+    return filter(lambda x: False, set([null(t)]))
```

Comment:
could also use `range` and map, not sure which is better for now. clearly we should add this to the scala side.